install.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/sh
+# TODO(everyone): Keep this script simple and easily auditable.
+
+set -e
+
+if [ "$(uname -m)" != "x86_64" ]; then
+  echo "Error: Unsupported architecture $(uname -m). Only x64 binaries are available." 1>&2
+  exit 1
+fi
+
+if [ "$OS" = "Windows_NT" ]; then
+  target="windows-386"
+  ext=".zip"
+  if ! command -v unzip >/dev/null; then
+    echo "Error: unzip is required to install coder-cli" 1>&2
+    exit 1
+  fi
+else
+  if ! command -v tar >/dev/null; then
```

Comment:
Why is this required? Won't it just exit out with command not found?

Reply:
Much cleaner to handle gracefully with a specific error message.

Reply:
How is it cleaner? It bloats the install script when the shell itself already handles it thanks to `set -e`.